pkg/codegen/dotnet/gen.go

```diff
@@ -481,7 +497,12 @@ func (pt *plainType) genInputProperty(w io.Writer, prop *schema.Property, indent
 
 		printComment(w, prop.Comment, indent)
 		fmt.Fprintf(w, "%s[Input(\"%s\"%s)]\n", indent, wireName, attributeArgs)
-		fmt.Fprintf(w, "%spublic %s %s { get; set; }%s\n", indent, propertyType, propertyName, initializer)
+		if prop.Secret {
+			// TODO: Implement secret constructor for basic types (currently only Map and Array are implemented)
+			panic("NYI: secret property for basic types")
```

Comment:
I was having trouble figuring out how to do this in C#, and it's not needed for my k8s use case.

AFAIK, nothing else is actually using the secret flag for now, but it would be ideal if we could figure out the correct code to generate here instead.

Reply:
Fixed